File: app/src/main/java/com/worldventures/dreamtrips/modules/common/delegate/LegalInteractor.java
package com.worldventures.dreamtrips.modules.common.delegate;

import com.worldventures.core.janet.SessionActionPipeCreator;
import com.worldventures.dreamtrips.modules.common.command.AcceptTermsCommand;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.techery.janet.ActionPipe;
import rx.schedulers.Schedulers;

@Singleton
public class LegalInteractor {

   private final ActionPipe<AcceptTermsCommand> termsPipe;

   @Inject
   public LegalInteractor(SessionActionPipeCreator sessionActionPipeCreator) {
      termsPipe = sessionActionPipeCreator.createPipe(AcceptTermsCommand.class, Schedulers.io());
   }

   public ActionPipe<AcceptTermsCommand> termsPipe() {
      return termsPipe;
   }
}

File: core/src/main/java/com/worldventures/core/modules/legal/LegalInteractor.kt
package com.worldventures.core.modules.legal

import com.worldventures.core.janet.SessionActionPipeCreator
import com.worldventures.core.modules.legal.command.AcceptTermsCommand
import com.worldventures.core.modules.legal.command.GetDocumentByTypeCommand
import rx.schedulers.Schedulers

class LegalInteractor(pipeCreator: SessionActionPipeCreator) {
   val acceptTermsPipe = pipeCreator.createPipe(AcceptTermsCommand::class.java, Schedulers.io())
   val getDocumentByTypePipe = pipeCreator.createPipe(GetDocumentByTypeCommand::class.java, Schedulers.io())
}
